src/utils/mod.rs

```diff
@@ -1377,12 +1376,115 @@ impl Default for PcrSelectSize {
 /// number of PCR required by the platform-specific
 /// specification with which the TPM is compliant or by the implementer if
 /// not adhering to a platform-specific specification.
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Default, Clone, PartialEq, Eq)]
 pub struct PcrSelections {
     size_of_select: PcrSelectSize,
     items: HashMap<HashingAlgorithm, BitFlags<PcrSlot>>,
 }
 
+impl PcrSelections {
+    /// Function for retrieiving the number of banks in the selection
+    pub fn len(&self) -> usize {
+        self.items.len()
+    }
+
+    /// Returns true if the selection is empty.
+    pub fn is_empty(&self) -> bool {
+        self.items.is_empty()
+    }
+
+    /// Returns the the elements present in `self` but not in `other` or None
+    /// if they are equal.
+    ///
+    /// # Arguments
+    ///
+    /// * `other` - A PcrSelections containing elements
+    ///             that will be removed from `self`.
+    ///     
+    ///
+    /// # Constraints
+    ///
+    /// * Cannot be called with `other` that contains elements that
+    ///   is not present in `self`.
+    ///
+    /// * Cannot be called with `other` that has a size_of_select
+    ///   that is different from the one in `self`.
+    ///
+    /// # Errors
+    ///
+    /// * Calling the method with `other` that contains elements
+    ///   not present in `self` will result in an InvalidParam error.
+    ///
+    /// * Calling the method with `other` that contains a size_of_select
+    ///   that is different then the one in `self` will result in
+    ///   a InvalidParam error.
+    ///
+    /// # Examples
+    /// ```
+    /// use tss_esapi::utils::{PcrSelectionsBuilder, PcrSlot, algorithm_specifiers::HashingAlgorithm};
+    /// // pcr selections
+    /// let pcr_selections = PcrSelectionsBuilder::new()
+    ///     .with_size_of_select(Default::default())
+    ///     .with_selection(HashingAlgorithm::Sha256, &[PcrSlot::Slot0, PcrSlot::Slot8])
+    ///     .build();
+    ///
+    /// // Another pcr selections
+    /// let other = PcrSelectionsBuilder::new()
+    ///     .with_size_of_select(Default::default())
+    ///     .with_selection(
+    ///         HashingAlgorithm::Sha256, &[PcrSlot::Slot0],
+    ///     )
+    ///     .build();
+    /// let diff = pcr_selections.difference(&other).unwrap();
+    /// assert_eq!(diff.unwrap().len(), 1);
+    /// ```
+    pub fn difference(&self, other: &Self) -> Result<Option<Self>> {
+        if self == other {
```

Comment:
That would not be needed if the `Option` is dropped: we would end up creating an empty `HashMap`.

Reply:
I kind of like that the function is returning an Option. Either there is a Some difference or there is None. But that is matter of taste and I think both works.

Reply:
Sure, if you prefer having an `Option` there, just go with it, that is totally fine with me 😃

Reply:
But if its going to change be a subtract then the option is not needed.

Reply:
Option is removed